File: assets/lua/meta/sweep.lua
--- @meta
--- 
--- @class sweep
sweep = {}

--- 
--- @param originA math.Vector
--- @param endA math.Vector
--- @param extentsA math.Vector
--- @param originB math.Vector
--- @param endB math.Vector
--- @param extentsB math.Vector
--- @param outResult bool
--- @param outTEntry number
--- @param outTExit number
--- @param outNormal math.Vector
--- @overload fun(originA: vector.Vector, endA: vector.Vector, extentsA: vector.Vector, originB: vector.Vector, endB: vector.Vector, extentsB: vector.Vector, outResult: bool, outTEntry: number, outTExit: number, outNormal: vector.Vector): 
function sweep.aabb_with_aabb(originA, endA, extentsA, originB, endB, extentsB, outResult, outTEntry, outTExit, outNormal) end

--- 
--- @param ext math.Vector
--- @param origin math.Vector
--- @param dir math.Vector
--- @param n math.Vector
--- @param d number
--- @param outResult bool
--- @param outT number
--- @overload fun(ext: vector.Vector, origin: vector.Vector, dir: vector.Vector, n: vector.Vector, d: number, outResult: bool, outT: number): 
function sweep.aabb_with_plane(ext, origin, dir, n, d, outResult, outT) end

File: core/shared/src/implementation/lua/libraries/sweep.cpp
// SPDX-FileCopyrightText: (c) 2019 Silverlan <opensource@pragma-engine.com>
// SPDX-License-Identifier: MIT
module;

module pragma.shared;

import :scripting.lua.libraries.sweep;

void Lua::sweep::AABBWithAABB(const Vector3 &originA, const Vector3 &endA, const Vector3 &extentsA, const Vector3 &originB, const Vector3 &endB, const Vector3 &extentsB, bool &outResult, float &outTEntry, float &outTExit, Vector3 &outNormal)
{
	outResult = pragma::math::sweep::aabb_with_aabb(originA, endA, extentsA, originB, endB, extentsB, &outTEntry, &outTExit, &outNormal);
}

void Lua::sweep::AABBWithPlane(const Vector3 &ext, const Vector3 &origin, const Vector3 &dir, const Vector3 &n, float d, bool &outResult, float &outT) { outResult = pragma::math::sweep::aabb_with_plane(origin, dir, ext, n, d, &outT); }
